shopinvader/services/settings.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2021 Akretion (http://www.akretion.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.addons.base_rest import restapi
+from odoo.addons.component.core import Component
+
+
+class ExportSettingsService(Component):
+    """Shopinvader service to expose allowed settings"""
+
+    _inherit = [
+        "base.shopinvader.service",
+    ]
+    _name = "shopinvader.settings.service"
+    _usage = "settings"
+    _description = __doc__
+
+    def get(self):
+        return {
+            **self._get_countries(),
+            **self._get_titles(),
+            **self._get_industries(),
+            **self._get_currencies(),
+            **self._get_languages(),
+        }
+
+    def _jsonify_fields_country(self):
+        return [
+            "name",
+            "code",
+            "id",
+        ]
+
+    def _get_countries(self):
+        countries = self.shopinvader_backend.allowed_country_ids.jsonify(
+            self._jsonify_fields_country()
+        )
+        return {"countries": countries}
+
+    @restapi.method(
+        [(["/countries"], "GET")],
+        auth="api_key",
```

Comment:
should be public.... Moreover, auth method `api_key` is no more provided by shopinvader. This authentication method is now provided by shopinvader_auth_api_key. This addon declare the REST controller used to serve all the services under the `shopinvader`root path with `api_key` as default auth. https://github.com/shopinvader/odoo-shopinvader/blob/14.0/shopinvader_auth_api_key/controllers/main.py The default auth is applied to all the service methods that doesn't explicitely declare the required authentication method. If you omit the `auth` param from `@restapi.method`, the default auth method defined on the controller registering the services at a given path will be applied. It's a key feature that allows you to serve the same services under 2 root paths with different authentication method. ex: with `shopinvader_auth_jwt` https://github.com/shopinvader/odoo-shopinvader/pull/1031/files#diff-65350ca569ae9b77446b7b6dacfd348e127ad4da850f96b4091a6434f4eaf8afR7

Reply:
I see, thank you for the explanation